src/postprocess/refine.jl

```diff
@@ -0,0 +1,183 @@
+# Refinement of some quantities of interest (density, forces) following the
+# strategy described in [CDKL2021].
+#
+# The overall strategy is to perform a first SCF,
+# transfer the result to a new basis set with a larger Ecut,
+# then perform an approximate first order correction in the larger space.
+#
+# To compute the first order correction ("refinement") δP in the larger space,
+# one needs to invert (Ω+K)δP = -R(P) where (Ω+K) is the Hessian matrix and
+# R(P) = [P, [P, H(P)]] is the residual.
+# See newton.jl for an explanation of these quantities.
+#
+# The basis elements of the smaller basis set are called low frequency (_1 suffix).
+# The additional basis elements are all high frequency (_2 suffix).
+# We approximate (Ω+K)_22 by M_22 which is cheap to compute and invert, and (Ω+K)_21 by 0.
```

Comment:
Here I would either rationalise in one sentence or refer again to the paper.

Reply:
Done